platform/src/generics/CMakeLists.txt

```diff
@@ -25,27 +26,32 @@ target_sources(RoughPy_Platform PRIVATE
         type_promotion.cpp
         value.cpp
         value_arithmetic.cpp
+        array.cpp
         value_compare.cpp
         value_math.cpp
 )
 
 
 add_subdirectory(builtin_types)
+add_subdirectory(mocking)
 add_subdirectory(multiprecision_types)
 add_subdirectory(polynomial_type)
 
 
 
 if (ROUGHPY_BUILD_TESTS)
-
-
     add_executable(test_generics
             test_value.cpp
+            test_array.cpp
     )
 
-    target_link_libraries(test_generics PRIVATE RoughPy::Platform GTest::gtest)
+    target_link_libraries(test_generics
+        PRIVATE
+            RoughPy::Platform
+            GTest::gtest
+            GTest::gmock
```

Comment:
I think the correct pattern is for RoughPy_Platform_Mocking library to (publicly) link GTest::gmock and for downstream users of Platform_Mocking to implilictly pick up that link.